generate_crossword/step_10/lib/providers.dart

```diff
@@ -0,0 +1,165 @@
+// Copyright 2024 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:built_collection/built_collection.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+import 'isolates.dart';
+import 'model.dart' as model;
+
+part 'providers.g.dart';
+
+const backgroundWorkerCount = 4;
+
+/// A provider for the wordlist to use when generating the crossword.
+@riverpod
+Future<BuiltSet<String>> wordList(WordListRef ref) async {
+  // This codebase requires that all words consist of lowercase characters
+  // in the range 'a'-'z'. Words containing uppercase letters will be
+  // lowercased, and words containing runes outside this range will
+  // be removed.
+
+  final re = RegExp('^[a-z]+\$');
```

Comment:
I think it's nice to use raw strings, since at least IntelliJ will then provide regex syntax highlighting :D

```suggestion
  final re = RegExp(r'^[a-z]+$');
```

Reply:
Can you tell that it's been a few years since I used IntelliJ?